### ISSUE TITLE
Question: Is it possible to run periodic lambdas in child account from a centralized account

### ISSUE THREAD

**Issue by rsandhu2** [capitalone/cloud-custodian#3155]
Hi, 
I was wondering if it is possible to kick off a periodic lambda deployed in a centralized account in various child accounts by invoking the periodic lambda with custom payload which includes the account id & region. 

Cross-account cloudwatch event support was implemented here and it works for cross-account cloudtrail events but this solution doesn't seem to be working for periodic lambdas: 
https://github.com/capitalone/cloud-custodian/pull/2005

Thanks in advance!

**Comment by kapilt** [comment]
the member-role does not work in periodic mode for centralized account, its limited to resource events at the moment. as your surmise in addition to the code change to enable this the event source (timer) would need manual deployment into each account (with bus event relaying) or be synthentically generated.

there's a separate option that folks have used previously by specifying a static assume_role in execution-options on the policy mode, but its static per account.

general purpose periodic centralized account exec is often better served by c7n-org, it handles large resource cardinalities better. folks have used it with fargate, eks, codebuild, jenkins, aws batch, ec2, etc for the same purpose with success.

**Comment by rsandhu2** [comment]
@kapilt member-role feature was added earlier this year for resource events. Is there a downside/technical limitations of adding the same feature for periodic events since account id exist in both?

**Comment by nic-b** [comment]
+1 for making it work for periodic.

I also note, that the handling of the session cache in utils.local_session() is different between cloudtrails and periodics. For cloudtrails, if member_role is set, the cache is reset every execution. For periodics though, this is is never done. Which means, if you decide to user periodic lambdas slightly differently than the default c7n installation (for master/child accounts), you can end up with a periodic lambda being called/reused multiple times, with different payloads, yet the connection (CONN_CACHE) persisting across executions and never being refreshed.

**Comment by kapilt** [comment]
c7n-org is the recommended solution, adding member-role support for periodics will get many users in trouble when it bombs because they do a policy on a resource type with high enough cardinality (ie. snapshots) that it won't ever fit in lambda execution run time limits.

**Comment by nic-b** [comment]
Good point about the lambda run time execution limits (although its now up to 15 mins I note, since October 2018). 

So the advice would be:
* For event based processing, use lambdas
* For bulk based ad hoc processing  use c7n-org

?
Would be great to see example setups of clients that have done this. At first glance, it feels a bit painful to maintain a timed run of c7n-org (either on prem, or bespoke ec2, or container) to handle the periodic use-case whilst still having lambda code for event based but maybe i'm missing some very simple work practices to handle this in a sustained way.

**Comment by kapilt** [comment]
yeah in a nutshell thats the best practice atm, event based for real time controls, and c7n or c7n-org for bulk scans, with c7n-org people setup in all kinda of setups, ec2, aws batch, fargate, codebuild, k8s are all common in production usage, basically they run it on a schedule either native or via a scheduling lambda, pull the policy repo(s) and execute the cli. that said i will support periodic mode centralized at some point, but with a doc caveat that its not going to work for scale usage with high cardinality counts (i've seen accounts with 6 & 7 figure resource counts, 15m isn't going to work there).

**Comment by nic-b** [comment]
Thanks for the reply. The scale comment at the end really adds context also
